packages/camlp-streams/camlp-streams.5.0/opam

```diff
@@ -32,7 +32,7 @@ bug-reports: "https://github.com/ocaml/camlp-streams/issues"
 license: "LGPL-2.1-only WITH OCaml-LGPL-linking-exception"
 depends: [
   "dune" {>= "2.5"}
-  "ocaml" {>= "4.02.3"}
+  "ocaml" {>= "5.0"}
```

Comment:
Was this change intended?

Reply:
Yes, as noted in the commit message: the 5.0 release doesn't work correctly with OCaml 4.x

Reply:
atdgen-runtime depends on camlp-streams, you just broke CI for a lot of folks out there.

Reply:
Sorry it's caused hassle, but what you had before was a broken library passing in CI!

Reply:
@dra27 oops, it wasn't in the PR message and I didn't see the commit message. Like @Lupus, I'm here because our CI broke following this package mutation that didn't change the package version, which defeated `opam lock`ing our dependencies to specific package versions.

Reply:
Thanks @samoht, I'll give `--ignore-constraints` a try!

Reply:
Our CI setup is somewhat custom, we used to do `opam install . --locked`, but for our large dependency tree it's too slow to build everything from scratch, played around with opam bin - didn't like it, too complex and fragile infrastructure. So we wrote a small tool that creates two switch export files - one with the most heavy dependencies (like core) and their transitive dependencies, and the other includes all project dependencies and their transitive dependencies as well. Projects are build in Docker with 2 distinct layers for importing switch states: base set of heavy dependencies, that change infrequently, and importing full set of dependencies on top of already imported base set. This way docker can effectively cache large amount of our projects OCaml depedencies. Works really well for us, fast and reliable.

Opam monorepo does not work with all packages in opam, and fails on some of our non-dune dependencies.